platform/darwin/src/MGLStyle.mm

```diff
@@ -287,6 +287,22 @@ - (void)removeImageForName:(NSString *)name
     self.mapView.mbglMap->removeImage([name UTF8String]);
 }
 
+- (MGLImage *)imageForName:(NSString *)name
+{
+    NSAssert(name, @"name is null");
+
+    auto spriteImage = self.mapView.mbglMap->getImage([name UTF8String]);
+
+    NSAssert(spriteImage, @"spriteImage is null");
+    if (!spriteImage)
+    {
+        return nil;
+    }
+
+    auto imageData = [NSData dataWithBytes:spriteImage->image.data.get() length:spriteImage->image.bytes()];
```

Comment:
You probably want to `PremultipliedImage` has a field called `data` that likely has what you want.